designs/FloatingVersions-In-PMUI.md

```diff
@@ -0,0 +1,104 @@
+
+### Current behavior
+
+### Updates Tab
+
+With the current behavior even if the user indicated a floating version, such as `*`, for the package version, the UI will tell the user that there is an update available which is misleading.
+
+#### Project File and Solution Explorer
+
+![SolutionExplorer](../resources/FloatingVersionsInPMUI/ProjectFile&SolutionExplorer.png)
+
+#### Project Manager UI
+
+![Package UI](../resources/FloatingVersionsInPMUI/PackageManagerUI.png)
+
+![Project UI](../resources/FloatingVersionsInPMUI/PackageManagerForSolution.png)
+
+```
+The requested and installed versions are not correct because dotnet list package shows a different installed version.
+```
+
+## Who are the customers
+
+All customers that use the Package Manager UI to install or update packages.
+
+## Goals
+
+* Allow users see the correct version of the package installed while using floating versions in the Package Manager UI in Project View.
+
+* Allow users to input floating versions to install in the Package Manager UI.
+
+* Allow users to distinguish between requested and resolved package versions in the Solution View.
+
+## Non-Goals
+
+* Any multi targeting improvements overall.
+
+## Solution
+
+### Installed Tab
+
+Display the correct resolved version of each package in this tab will correct the displayed version in this tab and avoid the package being listed in the `Updates Tab`.
+
+### Floating Version Display in Project View
+
+When searching for a package in the `Browse Tab` or `Installed Tab` the user is not able to select a floating or range version to install. We will implement a `Combo Box` that will autocomplete the available version of the package with the user input.
+
+This feature will enable the option `*` and ranges to be selected while installing/updating a package. Also enables scenarios beyond floating version and improves the UX.
+
+#### Floating Version
+
+![Floating Version](../resources/FloatingVersionsInPMUI/FloatingVersionInPMUI.png)
+
+#### Range Version
+
+![Range Version](../resources/FloatingVersionsInPMUI/RangesInPMUI.png)
+
+#### ComboBox
+
+![Combo Box](../resources/FloatingVersionsInPMUI/ComboBox1.png)
+
+![Combo Box Gif](../resources/FloatingVersionsInPMUI/ComboBoxGif.gif)
+
+### Floating Version Display in Solution View
+
+We will add a new column to show the requested version to match the behavior of `dotnet list package` command. Show correct highest resolved version in the solution in the package list view.
+
+![Solution View](../resources/FloatingVersionsInPMUI/SolutionView.png)
+
+## Future Work
+
+* Indicate the resolved version in the solution explorer.
```

Comment:
I believe this should be included in this spec as it completes it. After talking to users regarding floating versions, not many understand the wildcard syntax even when they use it. We should be very clear with what version is `requested` and what version is `resolved`.

Reply:
When we were doing the initial design spec, VS had a bug and this has been fixed, the solution explorer does show the resolved version. I will update the spec